File: Reverse_word_in_a_string_III/code.cpp
class Solution {
public:
    string reverseWords(string s) {
        int start = 0;
        int end;
        string res;
        char temp;

        for(int i=0; i<s.size(); i++){
            if(s[i] == ' '){
                end = i-1;
                for(int j=end; j>=start; j--){
                    temp = s[j];
                    res.append(1,temp);
                }
                res.append(" ");
                start = i+1;
            }
            if(i == s.size()-1){
                end = i;
                for(int j=end; j>=start; j--){
                    temp = s[j];
                    res.append(1,temp);
                }
            }
        }

        return res;
    }
};

File: Reverse_word_in_a_string_III/code.java
//User function Template for Java
class Solution
{
   
    String reverseWords(String str)
    {
        // your code here
        Stack<Character> s  = new Stack<>();
        StringBuilder sb = new StringBuilder();
        int n = str.length();
        
        for(int i=0;i<n;i++){
            
            if(str.charAt(i)!='.'){
                s.push(str.charAt(i));
            }
            else{
                while(!s.isEmpty()){
                    sb.append(s.pop());
                }
                 sb.append(".");
            }
        }
        
        while(!s.isEmpty()){
            sb.append(s.pop());
        }
        return sb.toString();
    }
}
